mpeg2/src/interleaving_muxer.rs

```diff
@@ -174,14 +193,14 @@ impl<W: Write> InterleavingMuxer<W> {
         let mut packets = vec![];
         self.streams.iter_mut().enumerate().for_each(|(stream_index, stream)| {
             while let Some(p) = stream.buffered_packets.pop_front() {
-                packets.push((stream_index, stream.last_written_ts, p));
+                packets.push((stream_index, p));
             }
             stream.last_written_ts = 0;
         });
 
-        packets.sort_unstable_by_key(|(_, ts, p)| p.dts_90khz.or(p.pts_90khz).unwrap_or(*ts));
-        for (stream_index, _, packet) in packets {
-            self.write_muxer_packet(stream_index, packet)?;
+        packets.sort_unstable_by_key(|(_, p)| p.0);
+        for (stream_index, packet) in packets {
+            self.write_muxer_packet(stream_index, packet.2.expect("should not fail"))?;
```

Comment:
The idea behind using `expect` over `unwrap` is that it encourages you to explain to the reader why it's okay to assume the value is `Some`. In this case, we can make the message a little more helpful:

```suggestion
            let packet = packet.2.expect("emit_packets should never leave None packets in the buffer");
            self.write_muxer_packet(stream_index, packet)?;
```